packages/block-library/src/categories/icon.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * WordPress dependencies
+ */
+import { SVG, Path } from '@wordpress/components';
```

Comment:
This is a good example of: How do we communicate in the manifest specific dependencies of the block (https://github.com/WordPress/gutenberg/pull/13693#issuecomment-464875602, `assets.json`, etc). 

We could have some NPM-style `dependencies` property in `block.json`, but we still have some challenges:

- How does WordPress resolve those?
- Or does it? (Do we ship something already bundled with dependencies resolved?)

I could imagine, for example, that the `icon` file referenced is one which already has everything bundled. This becomes very wasteful for shared dependencies, so I think we'd need some system there, but it's not clear to me how that looks in a way which is agnostic to the WordPress script loader / globals.

Reply:
i18n and assets are two issues we need to resolve to move forward. They are both very complex :)

I'm starting to question whether we should be splitting JS code into several files: `edit.js`, `save.js`, `icon.js`, `transforms.js`, etc. It only makes it harder to manage their dependencies without bigger benefits. If you want to insert a given block into the editor, you will have to load all those assets anyway.

It might be also be not enough to define only `renderCallback`, what if a plugin developer doesn't need a render callback but wants to expose REST API endpoint? How do they do it? What if plugin bundles multiple blocks into one JS file?